server/src/utils.rs

```diff
@@ -0,0 +1,14 @@
+use tower_lsp::lsp_types::Url;
+
+// This helper function converts a Url from tower_lsp into a string that can be used to
+// retrieve a file from the compilation state from slicec.
+pub fn convert_uri_to_slice_formated_url(uri: Url) -> Option<String> {
```

Comment:
A few suggestions for alternate names:
* `slicec_path_from_lsp_url`
*  `lsp_url_to_slicec_path`